src/xdp-utils.c

```diff
@@ -296,15 +299,154 @@ parse_app_info_from_flatpak_info (int pid, GError **error)
   return g_steal_pointer (&app_info);
 }
 
+static gboolean
+aa_is_enabled (void)
+{
+  static int apparmor_enabled = -1;
+  struct stat statbuf;
+  struct mntent *mntpt;
+  FILE *mntfile;
+
+  if (apparmor_enabled >= 0)
+    return apparmor_enabled;
+
+  apparmor_enabled = FALSE;
+
+  mntfile = setmntent ("/proc/mounts", "r");
+  if (!mntfile)
+    return FALSE;
+
+  while ((mntpt = getmntent (mntfile)))
+    {
+      g_autofree char *proposed = NULL;
+
+      if (strcmp (mntpt->mnt_type, "securityfs") != 0)
+        continue;
+
+      proposed = g_strdup_printf ("%s/apparmor", mntpt->mnt_dir);
+      if (stat (proposed, &statbuf) == 0)
+        {
+          apparmor_enabled = FALSE;
```

Comment:
I needed to change this to `TRUE` for the snap detection code to do anything.

Reply:
Oops.